test/auditing.jl

```diff
@@ -51,6 +51,58 @@
     end
 end
 
+@testset "Auditor - cxxabi selection" begin
+    for platform in (Linux(:x86_64; compiler_abi=CompilerABI(;cxxstring_abi=:cxx03)),
+                     Linux(:x86_64; compiler_abi=CompilerABI(;cxxstring_abi=:cxx11)))
+        # Look across multiple gcc versions; there can be tricksy interactions here
+        for gcc_version in (v"4", v"6", v"9")
+            # Do each build within a separate temporary directory
+            mktempdir() do build_path
+                libcxxstringabi_test = LibraryProduct("libcxxstringabi_test", :libcxxstringabi_test)
+                build_output_meta = autobuild(
+                    build_path,
+                    "libcxxstringabi_test",
+                    v"1.0.0",
+                    # Copy in the libfoo sources
+                    [build_tests_dir],
+                    # Easy build script
+                    raw"""
+                    cd ${WORKSPACE}/srcdir/cxxstringabi_tests
+                    make install
+                    install_license /usr/share/licenses/libuv/LICENSE
+                    """,
+                    # Build for this platform
+                    [platform],
+                    # The products we expect to be build
+                    [libcxxstringabi_test],
+                    # No depenedencies
+                    [];
+                    preferred_gcc_version=gcc_version
+                )
+
+                # Extract our platform's build
+                @test haskey(build_output_meta, platform)
+                tarball_path, tarball_hash = build_output_meta[platform][1:2]
+                @test isfile(tarball_path)
+
+                # Unpack it somewhere else
+                @test verify(tarball_path, tarball_hash)
+                testdir = joinpath(build_path, "testdir")
+                mkdir(testdir)
+                unpack(tarball_path, testdir)
+                prefix = Prefix(testdir)
+
+                # Ensure that the library detects as the correct cxxstring_abi:
```

Comment:
Maybe add also a unit test on `cppfilt` that checks it returns what is supposed to?